### PR TITLE
Add default toggle for confirm commands in config.yml

### PR DESCRIPTION
**Pull request by SupaHam**
This PR introduces a config property that controls the default prompting state for commands that require confirmation.

`default-disabled-confirm-commands` is a list of command names that when present, the command will not ask for confirmation by default. Since this list is empty by default, it means all confirming commands will be on by default. In the case that a command is present in the list, a player may manually use the appropriate confirm command to enable the feature for themselves.

Aims to fix issues including: #1667 and #1801

### CODE REVIEW

File: Essentials/src/com/earth2me/essentials/UserData.java
Comment (mdcfe) [inline Essentials/src/com/earth2me/essentials/UserData.java:904]: Why not use the default from `default-disabled-confirm-commands` here? It makes the `isPromptingPayConfirm` and `isPromptingClearConfirm` methods messy, and means we can just use the primitive `boolean`.

File: Essentials/src/config.yml
Comment (mdcfe) [inline Essentials/src/config.yml:521]: Why not just two config options, one for each command?

I'm not sure that having a list to disable just two confirmations is a more worthwhile approach than moving the whole confirmation system to `EssentialsCommand` and having a list to enable confirmations by default per command.

**Review by mdcfe — COMMENTED** [review]
See comments on diff. I'm not *against* this change, just it seems like there are more useful ways to do this.

File: Essentials/src/com/earth2me/essentials/UserData.java
Comment (SupaHam) [reply Essentials/src/com/earth2me/essentials/UserData.java:904]: Server reload. That config variable may change.

File: Essentials/src/config.yml
Comment (SupaHam) [reply Essentials/src/config.yml:521]: It's always easier to have new features commented out to enable the features for later versions of the config.

**Comment by Kakifrucht** [comment]
Looks good 👍

File: Essentials/src/config.yml
Comment (drtshock) [reply Essentials/src/config.yml:521]: Shouldn't these be disabled by default though and we can add them to have it enabled?

**Review by drtshock — CHANGES_REQUESTED** [review]

**Comment by SupaHam** [comment]
I replied to md's same comment

**Comment by mdcfe** [comment]
@SupaHam Not quite the same thing as what I said, as Trent's reply doesn't mention every command, just the two that already have confirmations implemented.

**Comment by SupaHam** [comment]
If they're disabled by default then they can't be enabled per update. A
similar implementation was done on another property before, iirc something
to do with signs.

**Comment by drtshock** [comment]
Ok then something needs to be done because we shouldn't be adding features that inconvenience users as much as this one can without leaving it disabled by default.

File: Essentials/src/com/earth2me/essentials/UserData.java
Comment (mdcfe) [reply Essentials/src/com/earth2me/essentials/UserData.java:904]: I don't see how that affects using it as the default in case the user config doesn't yet have that set. As far as I can see, that would be equivalent to what the PR does already, just much tidier and without having the config value retrieval itself split across two methods for each confirmation.

**Comment by mdcfe** [comment]
@drtshock Confirmations for `/pay` have been enabled by default for over a year, so it's not an addition since the last Spigot build. Perhaps clear confirmations could be disabled by default and pay could be enabled by default?

**Comment by SupaHam** [comment]
@drtshock the inconvenience was the lack of ability to disable the feature. Not the feature itself.

**Comment by Kakifrucht** [comment]
Waiting on @drtshock ?

**Review by mdcfe — APPROVED** [review]
I don't see anything holding this back any more.

**Review by DoNotSpamPls — APPROVED** [review]

**Review by Kakifrucht — APPROVED** [review]

**Review by drtshock — APPROVED** [review]

**Review by drtshock — CHANGES_REQUESTED** [review]
It's still enabled by default. I thought the conclusion of the conversation was that it should be disabled?

**Comment by mdcfe** [comment]
@drtshock @SupaHam I thought the conclusion was the opposite, because disabling it by default will break the behaviour that has been present for months?

**Comment by drtshock** [comment]
No, we're going back and fixing what shouldn't have been in the previous.

**Comment by Kakifrucht** [comment]
So your point is that before EssentialsX there was no payment confirmation and thus it should not be enabled by default? The problem with that is that there are different number formats and Essentials doesn't parse a ``,`` as decimal point and instead removes it. This caused issues in the past. Payment confirmation was added to prevent this issue. With this pull request it would become globally toggleable.

**Comment by drtshock** [comment]
Yes but it's still drastically changing functionality and will create a lot of support noise for us, as it already has.

**Comment by Kakifrucht** [comment]
So you are saying it should just be disabled by default? Don't know if that would cause less noise..

**Comment by mdcfe** [comment]
It won't create much more support noise than already exists, which is mostly +1s from people who know that this feature request exists. It's also going to be documented on the wiki as soon as it's merged, so I don't see that this is a worthwhile reason to undermine and obscure a useful feature.

Defaulting to disabled will also create inconsistent behaviour on servers that already use the feature - new users suddenly won't get confirmations after they join for the first time when they used to, and there's no way for server owners to easily realise this and change the EssentialsX config without constantly looking at EssentialsX's commit log when they update.

File: Essentials/src/com/earth2me/essentials/UserData.java
Comment (drtshock) [reply Essentials/src/com/earth2me/essentials/UserData.java:904]: Any thoughts on this supa?

**Comment by drtshock** [comment]
I still think this needs to be flipped. I think we should change the list to be `confirm-enabled-by-default` instead of disabled, and have the possible commands listed but commented out.

**Benefits**
* If we add confirms to new commands in the future, they will automatically be disabled by default. 
* Users upgrading will not be affected when they don't read the update log.
* Provides clarity to a currently confusing name.

Once we have this, I think we're ready for to pull @md678685's version change and push a release.

**Comment by SupaHam** [comment]
Sure thing.

**Comment by drtshock** [comment]
Merged with the changes I requested above.

**Pull request closed without merging**
